node/src/components/linear_chain/state.rs

```diff
@@ -92,9 +92,6 @@ impl LinearChain {
     // were waiting for that block.
     fn new_block(&mut self, block: &Block) -> Vec<Signature> {
         let signatures = self.collect_pending_finality_signatures(block.hash());
-        if signatures.is_empty() {
-            return vec![];
-        }
```

Comment:
Trying to understand this fix…
So this would now put an empty `BlockSignatures` into the cache. Where does that make a difference? (In `handle_finality_signature`, at least, it seems like it doesn't.)

Reply:
Yeah, the trick is that now we're caching empty `BlockSignatures` which later in `handle_is_bonded` will return `Some(_)` from `or_else(|| self.get_signatures)` method – which means that finality signature is for an already known block it's just that we don't have any signatures for it yet.

Reply:
It would be good to document that somewhere, maybe on the `signature_cache` field.

Reply:
Added in [`4b3ceea` (#1345)](https://github.com/CasperLabs/casper-node/pull/1345/commits/4b3ceea288596405a5977548ce5353b797c626a3).